include/deal.II/base/vectorization.h

```diff
@@ -3941,6 +3941,167 @@ operator<<(std::ostream &out, const VectorizedArray<Number, width> &p)
   return out;
 }
 
+
+/**
+ * @name Ternary operations on VectorizedArray
+ */
+//@{
+
+/**
+ * TODO
+ */
+enum SIMDComparison : int
+{
+  equal                 = _CMP_EQ_OQ,
+  not_equal             = _CMP_NEQ_OQ,
+  less_than             = _CMP_LT_OQ,
+  less_than_or_equal    = _CMP_LE_OQ,
+  greater_than          = _CMP_GT_OQ,
+  greater_than_or_equal = _CMP_GE_OQ
+};
+
+
+/**
+ * TODO
+ */
+template <SIMDComparison predicate, typename Number>
+DEAL_II_ALWAYS_INLINE inline Number
+compare_and_apply_mask(const Number &left,
+                       const Number &right,
+                       const Number &true_value,
+                       const Number &false_value)
+{
+  bool mask;
+
+  switch (predicate)
+    {
+      case SIMDComparison::equal:
+        mask = left == right;
+      case SIMDComparison::not_equal:
+        mask = left != right;
+      case SIMDComparison::less_than:
+        mask = left < right;
+      case SIMDComparison::less_than_or_equal:
+        mask = left <= right;
+      case SIMDComparison::greater_than:
+        mask = left > right;
+      case SIMDComparison::greater_than_or_equal:
+        mask = left >= right;
+    }
+
+  return mask ? true_value : false_value;
+}
+
+//@}
+
+#ifndef DOXYGEN
+#  if DEAL_II_COMPILER_VECTORIZATION_LEVEL >= 3 && defined(__AVX512F__)
+
+template <int predicate>
```

Comment:
It is for that reason that the SIMD overloads actually don't take an enum member as template argument, but an int. This allows to use one of the `unordered`, or `signalling` variants by simply providing the macro directly. For example `_CMP_LE_US` for the unorderd signalling variant instead of `SIMDComparison::less_than_or_equal`. (All of this makes a difference only if a `NaN` is encountered in the `VectorizedArrays`s.)